File: solutions/rust/src/2418/sort.rs
use std::iter::FromIterator;

impl Solution {
    pub fn sort_people(names: Vec<String>, heights: Vec<i32>) -> Vec<String> {
        let mut vec = Vec::from_iter(names.into_iter().zip(heights.into_iter()));
        vec.sort_by_key(|(name, h)| -*h);
        return vec.into_iter().map(|(name, h)| name).collect();
    }
}

File: solutions/javascript/src/2418/sort.js
/**
 * @param {string[]} names
 * @param {number[]} heights
 * @return {string[]}
 */
var sortPeople = function (names, heights) {
  return new Array(names.length)
    .fill()
    .map((_, i) => i)
    .sort((a, b) => heights[b] - heights[a])
    .map((i) => names[i]);
};
